cli_037_smoke_plumes_US/contents/src/__init__.py

```diff
@@ -36,125 +37,254 @@
+        logging.info('Fetching existing IDs')
+        r = cartosql.getFields(id_field, table, f='csv', post=True, user=CARTO_USER, key=CARTO_KEY)
+        # turn the response into a list of strings, removing the first and last entries (header and an empty space at end)
+        return r.text.split('\r\n')[1:-1]
+    else:
+        # if the table does not exist, create it with columns based on the schema input
+        logging.info('Table {} does not exist, creating'.format(table))
+        cartosql.createTable(table, schema, user=CARTO_USER, key=CARTO_KEY)
+        # if a unique ID field is specified, set it as a unique index in the Carto table; when you upload data, Carto
+        # will ensure no two rows have the same entry in this column and return an error if you try to upload a row with
+        # a duplicate unique ID
+        if id_field:
+            cartosql.createIndex(table, id_field, unique=True, user=CARTO_USER, key=CARTO_KEY)
+        # if a time_field is specified, set it as an index in the Carto table; this is not a unique index
+        if time_field:
+            cartosql.createIndex(table, time_field, user=CARTO_USER, key=CARTO_KEY)
+        # return an empty list because there are no IDs in the new table yet
+        return []
+
+'''
+FUNCTIONS FOR THIS DATASET
+
+The functions below have been tailored to this specific dataset.
+They should all be checked because their format likely will need to be changed.
+'''
+
 # Generate UID
 def genUID(date, pos_in_shp):
+    '''Generate unique id using date and feature index in retrieved GeoJSON
+    INPUT   date: date for which we want to generate id (string)
+            pos_in_shp: index of the feature in GeoJSON (integer)
+    RETURN  unique id of the feature in GeoJSON (string)
+    '''
     return str('{}_{}'.format(date, pos_in_shp))
 
-
 def getDate(uid):
-    '''first 8 chr of ID'''
+    '''
+    Split uid variable using '_' to get the first eight elements which represent the date 
+    INPUT   uid: unique ID that we already have in our Carto table (string)
+    RETURN  date of the feature in GeoJSON (string)
+    '''
     return uid.split('_')[0]
 
-
 def formatObservationDatetime(start, end, datetime_format=DATETIME_FORMAT):
+    '''
+    Reformat the start and end date according to DATETIME_FORMAT
+    INPUT   start: start date of smoke plume observation (string)
+            end: end date of smoke plume observation (string)
+            datetime_format: format of date in source shapefile (string)
+    RETURN  start date, end date and duration of the observation (strings)
+    '''
+    # split the start date to separate out date and time
     date, time = start.split(' ')
+    # get year of start date from first four index of date
     year = int(date[:4])
-    day = int(date[4:])-1 # Account for fact that we're initiating from day 1
+    # get fourth and the following indexes from date and subrtract 1 to get day
+    # to account for fact that we're initiating from day 1
```

Comment:
```suggestion
    # 1 is subtracted because we will add the day number to January 1 to get the date. The source starts with January 1 having a day number of 1, so we would want to add 0 to January 1 to get the correct date.
```